pkg/registry/common/dnsresolve/common.go

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnsresolve
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+)
+
+// Domain represents wanted domain and service
+type Domain struct {
+	// Name is name of wanted domain
+	Name string
+	// Service is name of wanted service
+	Service string
+}
+
+// Resolver is DNS resolver
+type Resolver interface {
```

Comment:
Why do we have a new Resolver interface, why not just use the standard net.* ones?

Reply:
Ah, I see... testing shims...

Reply:
@edwarnicke Note: standard net.* provides structures